app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -151,32 +150,34 @@ class FormController(
 
                 iteration.checkYourAnswers match {
                   case Some(checkYourAnswers) if checkYourAnswers.sectionNumber == sectionNumber =>
-                    val visibleIteration = iteration.copy(singletons =
-                      NonEmptyList.fromListUnsafe(
-                        iteration.singletons.filter(s =>
-                          formModelOptics.formModelVisibilityOptics.formModel.availableSectionNumbers
-                            .contains(s.sectionNumber)
-                        )
+                    val maybeVisibleIteration: Option[Bracket.AddToListIteration[Visibility]] =
+                      formModelOptics.formModelVisibilityOptics.formModel.addToListBrackets.collectFirst {
```

Comment:
Can't you use `def bracket(sectionNumber: SectionNumber): Bracket[A]` defined on `FormModel`?